modules/dataframe/include/inviwo/dataframe/datastructures/dataframe.h

```diff
@@ -221,11 +220,9 @@ std::shared_ptr<TemplateColumn<T>> DataFrame::addColumn(const std::string &heade
 }
 
 template <typename T>
-std::shared_ptr<TemplateColumn<T>> DataFrame::addColumn(const std::string &header,
-                                                        const std::vector<T> &data) {
-    auto col = std::make_shared<TemplateColumn<T>>(header);
-    col->getTypedBuffer()->getEditableRAMRepresentation()->getDataContainer() = data;
-    return col;
+auto DataFrame::addColumn(const std::string &header, const std::vector<T> data) {
```

Comment:
don't make it const, then you cant move from it....

Reply:
Interesting! I feel like the compiler could mention that...

Reply:
just, unfortunately it just silently falls back to doing a copy :(